core/src/main/java/org/elasticsearch/index/query/HasParentQueryParser.java

```diff
@@ -30,9 +30,10 @@
 public class HasParentQueryParser implements QueryParser<HasParentQueryBuilder>  {
 
     private static final HasParentQueryBuilder PROTOTYPE = new HasParentQueryBuilder("", EmptyQueryBuilder.PROTOTYPE);
-    private static final ParseField QUERY_FIELD = new ParseField("query", "filter");
-    private static final ParseField SCORE_FIELD = new ParseField("score_mode").withAllDeprecated("score");
-    private static final ParseField TYPE_FIELD = new ParseField("parent_type", "type");
+    public static final ParseField QUERY_FIELD = new ParseField("query", "filter");
+    public static final ParseField SCORE_MODE_FIELD = new ParseField("score_mode").withAllDeprecated("score");
```

Comment:
Hmm - I believe didn't change that line, just checked, this https://github.com/elastic/elasticsearch/commit/d2e53e0e0c1c20f1be178a1da8bc76c9e5993cf8#diff-838f43c5421ea0e3f74f4548fddba923R50 is where it comes from.

So - looked closer at the implementation of ParseField. This thing is weird: When switching to the version you proposed this would change the behaviour at least for getPreferredName() which we use for printing the json:


    System.out.println((new ParseField("score_mode").withAllDeprecated("score"))getPreferredName());

Gives score_mode, while

    System.out.println((new ParseField("score", "score_mode")).getPreferredName());

gives score. I'm confused. Looking further what we actually want here.